tools/update_snapcraft.py

```diff
@@ -0,0 +1,146 @@
+    return (
+        f"{path.parent.stem}-last"
+        if f"{path.parent.stem}-last" in revisions
+        else max(revisions, key=cmp_to_key(semver.compare))
+    )
+
+
+def get_latest_tempest_revision(release):
+    """Return the requirements entry for the latest tempest revision."""
+    return get_latest_revision_from_release_file(
+        RELEASES_REPO_PATH / "deliverables" / release / "tempest.yaml"
+    )
+
+
+def get_latest_plugin_requirements(release):
+    """Return list of requirements entries for the latest tempest plugin revisions."""
+    result = []
+    plugin_release_file_paths = [
+        entry
+        for entry in (RELEASES_REPO_PATH / "deliverables" / release).iterdir()
+        if entry.is_file() and yaml.safe_load(entry.read_text())["type"] == "tempest-plugin"
+    ]
+    for path in plugin_release_file_paths:
+        try:
+            latest_revision = get_latest_revision_from_release_file(path)
+            result.append(OPENSTACK_REPO_URL_FMT.format(project=path.stem, ref=latest_revision))
+        except KeyError as exception:
+            logger.warning("Skipping path:[%s], error:[%s]", path, repr(exception))
+    return sorted(result)
+
+
+def get_latest_tempestconf_requirements():
+    """Return the requirements entry for the latest python-tempestconf revision."""
+    project = "python-tempestconf"
+    latest_revision = feedparser.parse(PYPI_RSS_FEED_FMT.format(project=project)).entries[0].title
+    return OPENINFRA_REPO_URL_FMT.format(project=project, ref=latest_revision)
+
+
+def clone_releases_repository(reuse):
+    """Clone the OpenStack releases repository.
+
+    If the reuse flag is given, try to reuse the existing repository.
+    """
+    if not RELEASES_REPO_PATH.exists():
+        logger.info("Cloning releases repository")
+        pygit2.clone_repository(RELEASES_REPO_URL, RELEASES_REPO_PATH)
+    elif reuse:
+        repo_path = pygit2.discover_repository(RELEASES_REPO_PATH)
+        if not repo_path:
+            raise RuntimeError(
+                f"Reuse flag given however {RELEASES_REPO_PATH} is not a git repository."
+            )
+        if pygit2.Repository(repo_path).remotes["origin"].url != RELEASES_REPO_URL:
+            raise RuntimeError(
+                f"Repository: {repo_path} does not list {RELEASES_REPO_URL} as origin."
+            )
+        logger.info("Using the repository at %s", repo_path)
+    else:
+        raise RuntimeError(f"Reuse flag not given and {RELEASES_REPO_PATH} exists.")
+
+
+def main(args):
+    """Entry point to the application."""
+    clone_releases_repository(args.reuse)
+    snapcraft_yaml_path = Path(__file__).parent.parent / "snap" / "snapcraft.yaml"
+    snapcraft_yaml = yaml.safe_load(snapcraft_yaml_path.read_text())
+
+    snapcraft_yaml["parts"]["tempest"]["source-tag"] = get_latest_tempest_revision(args.release)
+    snapcraft_yaml["parts"]["tempest"]["python-packages"] = [
+        "confluent-kafka==1.8.2",
```

Comment:
Can we add this to some constant, so it will be easier to change it.

Reply:
do you mean the version or the complete string? can you give an example of what would you like to see here?

Reply:
something like this
```python
ADDITIONAL_REQUIREMENTS = [
    "confluent-kafka==1.8.2",
]
```
or if there will no more requirements like this, than use simply
```python
CONFLUENT_KAFKA_VERSION="1.8.2"
```

Reply:
ok will update.